drivers/serial/uart_stm32.c

```diff
@@ -1786,16 +1786,24 @@ static int uart_stm32_pm_action(const struct device *dev,
 		/* Stop device clock. Note: fixed clocks are not handled yet. */
 		err = clock_control_off(data->clock, (clock_control_subsys_t)&config->pclken[0]);
 		if (err != 0) {
-			LOG_ERR("Could not enable (LP)UART clock");
+			/*
+			 * Warn without console output when going to sleep,
+			 * "Could not enable (LP)UART clock"
+			 * but don't block PM suspend
+			 */
 			return err;
 		}
 
 		/* Move pins to sleep state */
 		err = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_SLEEP);
-		if (err == -ENOENT) {
-			/* Warn but don't block PM suspend */
-			LOG_WRN("(LP)UART pinctrl sleep state not available ");
-		} else if (err < 0) {
+		if ((err < 0) && (err != -ENOENT)) {
+			/*
+			 * If returning -ENOENT, no pins wher defined for sleep mode :
+			 * Warn without console output when going to sleep,
+			 * "(LP)UART pinctrl sleep state not available"
+			 * but don't block PM suspend.
+			 * Else return the error.
```

Comment:
I agree with the change: 
Logging in -ENOENT case requires console which leads to an issue as we're in the process of falling asleep. Console may not be available and we don't want to wake up the system.

However, the comment is confusing as it doesn't describe what we're doing (no if/else is done, no "warn w/o console" is done neither).